nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchHttp.java

```diff
@@ -227,7 +226,10 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         List<FlowFile> flowFilesToTransfer = new LinkedList<>(flowFiles);
 
         final StringBuilder sb = new StringBuilder();
-        final String baseUrl = trimToEmpty(context.getProperty(ES_URL).evaluateAttributeExpressions().getValue());
+        final String baseUrl = context.getProperty(ES_URL).evaluateAttributeExpressions().getValue().trim();
+        if (StringUtils.isEmpty(baseUrl)) {
```

Comment:
if baseUrl is empty should the exception message be that "Elasticsearch URL evaluates to empty"  or something?  Your message is always going to be "... not valid: " since baseUrl will be empty.

Reply:
D'oh, good catch, I was doing it differently before and left in the useless error message :P Will change in both places.